src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -55,64 +58,116 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-    throws ServletException, IOException {
-    String stringToMatch = request.getParameter("searchterm");
-    stringToMatch = trimAndRemoveCommas(stringToMatch);
-    request.setAttribute("searchQuery", stringToMatch);
-
-    List<Recipe> recipesListToReturn = recipesMatching(
-      request,
-      response,
-      stringToMatch
+    throws ServletException, IOException, NullPointerException {
+    String searchTerm = getParameter("searchterm", request);
+    String difficulty = getParameter("difficulty", request);
+    String time = getParameter("time", request);
+
+    request.setAttribute("searchTerm", searchTerm);
+    request.setAttribute("difficulty", difficulty);
+    request.setAttribute("prepTime", time);
+
+    searchTerm = sanitizeString(searchTerm);
+    difficulty = sanitizeString(difficulty);
+    request.setAttribute(
+      "difficulty",
+      difficulty
     );
```

Comment:
You're doing this already in line 67. Please do the sanitization in the same line (inside the setAttribute) if that's the right thing.

Reply:
I forgot to remove the duplication of setAttribute function usage.